tools/deb/istio.mk

```diff
@@ -0,0 +1,89 @@
+.PHONY: deb/build-in-docker sidecar.deb
+
+# Make the deb image using the CI/CD image and docker, for users who don't have 'fpm' installed.
+# TODO: use 'which fpm' to detect if fpm is installed on host, consolidate under one target ('deb')
+deb/build-in-docker:
+	(cd ${TOP}; docker run --rm -u $(shell id -u) -it \
+        -v ${GO_TOP}:${GO_TOP} \
+        -w ${PWD} \
+        -e USER=${USER} \
+        -e GOPATH=${GOPATH} \
+		--entrypoint /usr/bin/make ${CI_HUB}/ci:${CI_VERSION} \
+		deb )
+
+# Create the 'sidecar' deb, including envoy and istio agents and configs.
+# This target uses a locally installed 'fpm' - use 'docker.sidecar.deb' to use
+# the builder image.
+# TODO: consistent layout, possibly /opt/istio-VER/...
+sidecar.deb: ${OUT}/istio-sidecar.deb
+
+deb: ${OUT}/istio-sidecar.deb
+
+ISTIO_DEB_SRC=tools/deb/istio-start.sh \
```

Comment:
An unfortunate design of make is that = is more natural for people to use but := is generally what people should use (to avoid make wasting time trying to do variable expansion on each use).  Eventually I or someone else will probably do a find/replace to make most = be := .

Reply:
To be honest I didn't even know about this - so := is not making any expansion ? Is there a quick doc, the make manual is a pain...

Reply:
The right-side of := is expanded to compute the result for the left-side, but once the left-side is computed it remains static (unlike = which uses dynamic evaluation on each use of the variable).  In the following example "make example" produces the output "hello world".

FIRST:=world
SECOND=$(FIRST)
THIRD:=$(FIRST)

FIRST:=hello

example:
        @echo $(SECOND) $(THIRD)